lib/Common/Exceptions/ExceptionCheck.h

```diff
@@ -94,17 +94,17 @@ class AutoFilterExceptionRegion
 #define AUTO_NO_EXCEPTION_REGION
 #endif
 
-#if DBG
-class DebugCheckNoException
+#ifdef EXCEPTION_CHECK
+class CheckNoException
 {
 public:
-    DebugCheckNoException() : hasException(true) { SAVE_EXCEPTION_CHECK;}
-    ~DebugCheckNoException() { Assert(!hasException); RESTORE_EXCEPTION_CHECK; }
+    CheckNoException() : hasException(true) { SAVE_EXCEPTION_CHECK;}
+    ~CheckNoException() { Assert(!hasException); RESTORE_EXCEPTION_CHECK; }
```

Comment:
If this is available in non-debug builds, should this be AssertOrFailFast?

Reply:
Reverted (because not using it in LEAVE_SCRPT_START_EX and LEAVE_SCRIPT_IF_ACTIVE anymore)

---
In reply to: [196615914](https://github.com/Microsoft/ChakraCore/pull/5335#discussion_r196615914) [](ancestors = 196615914)